root/etc/cont-init.d/50-config

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/with-contenv bash
+
+# create folders
+mkdir -p \
+    /config/.ssh
+
+USER_NAME=${USER_NAME:-linuxserver.io}
+
+# set password for abc to unlock it and set sudo access
+sed -i "/${USER_NAME} ALL.*/d" /etc/sudoers
+if [ "$SUDO_ACCESS" == "true" ]; then
+    if [ -n "$SUDO_PASSWORD" ] || [ -n "$SUDO_PASSWORD_FILE" ]; then
+        echo "${USER_NAME} ALL=(ALL) ALL" >> /etc/sudoers
+    else
+        echo "${USER_NAME} ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+    fi
+fi
+
+[[ -n "$SUDO_PASSWORD_FILE" ]] && \
+    SUDO_PASSWORD=$(cat "$SUDO_PASSWORD_FILE")
+SUDO_PASSWORD=${SUDO_PASSWORD:-$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${1:-8};echo;)}
+echo "${USER_NAME}:${SUDO_PASSWORD}" | chpasswd
+
```

Comment:
Awesome! Might I suggest checking that `$SUDO_PASSWORD_FILE` is a readable file rather than just not empty? (i.e. use `[ -r ]`)
I would be very confused if i set a password file and then got a totally different random password because I tyop'd the path :)

Not sure if you guys usually do error output from this config file but you could print something if its not readable

Reply:
@kaysound
`-r` only checks read permissions. How about `-f` to make sure it's a file and not a folder? Can you check if docker secrets pass that test?

Reply:
I just tried my mariadb PR with `[ -f "$MYSQL_ROOT_PASSWORD_FILE" -a -r "$MYSQL_ROOT_PASSWORD_FILE" ]` (checks for readable, regular file) and it works with secrets

Reply:
Thanks